graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -104,17 +104,25 @@ object PageRank extends Logging {
       graph: Graph[VD, ED], numIter: Int, resetProb: Double = 0.15,
       srcId: Option[VertexId] = None): Graph[Double, Double] =
   {
+    val personalized = srcId isDefined
     // Initialize the PageRank graph with each edge attribute having
-    // weight 1/outDegree and each vertex with attribute 1.0.
+    // weight 1/outDegree and each vertex with attribute resetProb.
+    // When running personalized pagerank, only the source vertex
+    // has an attribute resetProb. All others are set to 0.
     var rankGraph: Graph[Double, Double] = graph
       // Associate the degree with each vertex
       .outerJoinVertices(graph.outDegrees) { (vid, vdata, deg) => deg.getOrElse(0) }
       // Set the weight on the edges based on the degree
       .mapTriplets( e => 1.0 / e.srcAttr, TripletFields.Src )
       // Set the vertex attributes to the initial pagerank values
-      .mapVertices( (id, attr) => resetProb )
+      .mapVertices( (id, attr) => {
+        if (personalized) {
+          if (id == srcId.get) resetProb else 0.0
+        } else {
+          resetProb
+        }
+      })
 
-    val personalized = srcId isDefined
     val src: VertexId = srcId.getOrElse(-1L)
```

Comment:
Move up `val src: VertexId = srcId.getOrElse(-1L) ` to `val personalized = srcId isDefined`